apps/omg_watcher/lib/exit_processor/core.ex

```diff
@@ -727,6 +1033,35 @@ defmodule OMG.Watcher.ExitProcessor.Core do
   defp competitor_for(signed1, %Transaction.Signed{raw_tx: known_raw_tx}),
     do: competitor_for(signed1, known_raw_tx)
 
+  @spec get_double_spends(tx_input_info, tx_input_info) :: [{0..3, Utxo.Position.t(), 0..3}]
+        when tx_input_info: Transaction.Signed.t() | Transaction.t() | KnownTx.t() | [{Transaction.Input.t(), 0..3}]
+  defp get_double_spends(inputs, known_spent_inputs) when is_list(inputs) and is_list(known_spent_inputs) do
+    # TODO: possibly ineffective if Transaction.max_inputs >> 4
+    list =
+      for {left, left_index} <- inputs,
+          {right, right_index} <- known_spent_inputs,
+          left == right,
+          left_index <= right_index,
+          do: {left_index, left, right_index}
+
+    :lists.usort(list)
+  end
+
+  defp get_double_spends(inputs, known_spent_inputs) do
+    get_double_spends(index_inputs(inputs), index_inputs(known_spent_inputs))
+  end
+
+  defp index_inputs(inputs_list) when is_list(inputs_list) do
+    inputs_list
+  end
+
+  defp index_inputs(%Transaction{} = raw_tx) do
+    Transaction.get_inputs(raw_tx) |> Enum.filter(&Utxo.Position.non_zero?/1) |> Enum.with_index()
```

Comment:
at the moment, when "gaps" in inputs are allowed, you should do this Enuming in swapped order - first apply indices then filter out.

At one point when we decide it's safe for `Transaction.get_inputs` to filter non-zeros always, we'll just ditch the filtering in here and `with_index` will be in the right spot anyway

Reply:
done